### PR TITLE
Add JacksonSdkLiteralType

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Add a new implementation for SdkLiteralType: JacksonSdkLiteralType is able to take bean-like classes as literal types

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 We don't support arbitrary AutoValues as SdkLiteralTypes, but this PR fixes this for Java. Now Bean-like classes (Autovalues, JacksonBinding annotated classes, and the ones that follow the Java Bean spec) can be literal types.

These bean objects will be translated to SDK Structs.

## Tracking Issue
_NA_

## Follow-up issue
We need to support the same for Scala

### CODE REVIEW

File: flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    Kind valueKind = value.kind();
+    switch (valueKind) {
+      case STRING_VALUE:
+        jsonGenerator.writeString(value.stringValue());
+        break;
+      case BOOL_VALUE:
+        jsonGenerator.writeBoolean(value.boolValue());
+        break;
+      case NUMBER_VALUE:
+        jsonGenerator.writeNumber(value.numberValue());
```
Comment (andresgomezfrr) [inline flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java:59]: Can it be any number? Double or Long? It is different than the SimpleType, right?

File: flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      serializeValue(entry.getValue(), jsonGenerator, serializerProvider);
+    }
+    jsonGenerator.writeEndObject();
+  }
+
+  private void serializeValue(
+      Value value, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+      throws IOException {
+    Kind valueKind = value.kind();
+    switch (valueKind) {
```
Comment (andresgomezfrr) [inline flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java:51]: Are the map values no supported inside a struct?

File: flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/SdkLiteralTypeModule.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+import org.flyte.api.v1.Literal;
+import org.flyte.flytekit.jackson.deserializers.LiteralStructDeserializer;
+import org.flyte.flytekit.jackson.serializers.StructSerializer;
+
+class SdkLiteralTypeModule extends Module {
+
+  @Override
+  public String getModuleName() {
+    return "test";
```
Comment (andresgomezfrr) [inline flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/SdkLiteralTypeModule.java:31]: I think that you prefer another name here :)

Comment (narape) [reply flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/SdkLiteralTypeModule.java:31]: yes I missed this

File: flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java (hunk 1 above)
Comment (narape) [reply flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java:59]: Yes it can, Jackson is very lenient in deserializing tree nodes into the autovalues. You can check the unit tests to see it in action, it supports the long and the double

File: flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java (hunk 2 above)
Comment (narape) [reply flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java:51]: I will check further. Structs in flyte are really [google proto structs](https://protobuf.dev/reference/protobuf/google.protobuf/#struct) I think maps are meant to be embedded structs but I will make a test for it

Comment (narape) [reply flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/serializers/StructSerializer.java:51]: I added tests that use maps in the autovalues. Jackson can seamlessly translate from maps to structs and back

**Review by andresgomezfrr — APPROVED** [review]
LGTM
